cmd/nginx/main.go

```diff
@@ -240,14 +235,20 @@ func handleFatalInitError(err error) {
 		err)
 }
 
-func registerHandlers(enableProfiling bool, port int, ic *controller.NGINXController, mux *http.ServeMux) {
+func registerHandlers(
+	enableProfiling bool,
+	port int,
+	ic *controller.NGINXController,
+	mux *http.ServeMux,
+	reg *prometheus.Registry) {
+
 	// expose health check endpoint (/healthz)
 	healthz.InstallHandler(mux,
 		healthz.PingHealthz,
 		ic,
 	)
 
-	mux.Handle("/metrics", promhttp.Handler())
+	mux.Handle("/metrics", promhttp.HandlerFor(reg, promhttp.HandlerOpts{}))
```

Comment:
Handler() will instrument the handler too. If you want to keep this, use this instead:
```
       http.Handle(
               "/metrics",
               promhttp.InstrumentMetricHandler(
                       prometheusRegistry,
                       promhttp.HandlerFor(prometheusRegistry, promhttp.HandlerOpts{}),
               ),
       )
```

Reply:
@discordianfish `InstrumentMetricHandler` is not available in 0.8. If safe to use prometheus/client_golang master?